### PR TITLE
Add shims for Spark 3.2.3

### LINKED ISSUE #6024: [FEA] Add support for Spark 3.2.3 SNAPSHOT
**Is your feature request related to a problem? Please describe.**
Spark 3.2.2 is released so the branch-3.2 will now be Spark 3.2.3, we should add a shim layer to support it.

### PR DESCRIPTION
**Pull request by firestarman**
closes #6024

Signed-off-by: Firestarman <firestarmanllc@gmail.com>

### CODE REVIEW

**Comment by firestarman** [comment]
Spark 3.2.3 release is in progress, now use 3.2.3-SNAPSHOT jars to build.
Need to remove "SNAPSHOT" from the version string once the release is done.

**Comment by firestarman** [comment]
build

**Comment by firestarman** [comment]
build

**Comment by firestarman** [comment]
IT verification is being blocked by an issue same as [this one](https://blossom.nvidia.com/sw-gpu-spark-jenkins/blue/organizations/jenkins/rapids_databricks_nightly-pre_release-github/detail/rapids_databricks_nightly-pre_release-github/215/pipeline/124).

I will try to run it per file locally and file following issues once I find any faling test.

File: sql-plugin/src/main/323/scala/com/nvidia/spark/rapids/shims/RegExpShim.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims
+
+object RegExpShim {
+  // Handle regexp_replace inconsistency from https://issues.apache.org/jira/browse/SPARK-39107
+  def reproduceEmptyStringBug(): Boolean = false
```
Comment (gerashegalov) [inline sql-plugin/src/main/323/scala/com/nvidia/spark/rapids/shims/RegExpShim.scala:21]: This is a fourth occurrence of this shim object with the same impl: 314, 322, 323, 330+. Can we create a new dir `post314` and include it in
in the minimum amount of sparkXYZ.pattern and sparkXYZ.sources.

File: sql-plugin/src/main/323/scala/com/nvidia/spark/rapids/shims/SparkShims.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  override def getSparkShimVersion: ShimVersion = ShimLoader.getShimVersion
+
+  override def getFileScanRDD(
+      sparkSession: SparkSession,
+      readFunction: PartitionedFile => Iterator[InternalRow],
+      filePartitions: Seq[FilePartition],
+      readDataSchema: StructType,
+      metadataColumns: Seq[AttributeReference]): RDD[InternalRow] = {
+    new FileScanRDD(sparkSession, readFunction, filePartitions)
+  }
```
Comment (gerashegalov) [inline sql-plugin/src/main/323/scala/com/nvidia/spark/rapids/shims/SparkShims.scala:41]: Thim method looks the same for 

- Spark31XShims (311until320-nondb)
- 320, 321, 321cdh, 322, 323

Can we find a way to consolidate some?

File: sql-plugin/src/main/323/scala/com/nvidia/spark/rapids/shims/RegExpShim.scala (hunk 1 above)
Comment (firestarman) [reply sql-plugin/src/main/323/scala/com/nvidia/spark/rapids/shims/RegExpShim.scala:21]: I plan to move this `reproduceEmptyStringBug()` to `SparkShimImpl`, and remove all the files named `RegExpShim.scala` in a following PR.

Comment (firestarman) [reply sql-plugin/src/main/323/scala/com/nvidia/spark/rapids/shims/RegExpShim.scala:21]: `post314` apears to cover 320, 321, but they have a different impl.

**Review by gerashegalov — APPROVED** [review]
LGTM,  would be good to reduce duplication in follow up PRs

File: sql-plugin/src/main/323/scala/com/nvidia/spark/rapids/shims/RegExpShim.scala (hunk 1 above)
Comment (gerashegalov) [reply sql-plugin/src/main/323/scala/com/nvidia/spark/rapids/shims/RegExpShim.scala:21]: I used post314 to distinguish it from 314+. Similar how we with pre320-treenode and post320-treenode

File: sql-plugin/src/main/323/scala/com/nvidia/spark/rapids/shims/SparkShims.scala (hunk 2 above)
Comment (firestarman) [reply sql-plugin/src/main/323/scala/com/nvidia/spark/rapids/shims/SparkShims.scala:41]: We may have a `Spark32XShims `, will try it in a follow up PR.

File: sql-plugin/src/main/323/scala/com/nvidia/spark/rapids/shims/RegExpShim.scala (hunk 1 above)
Comment (firestarman) [reply sql-plugin/src/main/323/scala/com/nvidia/spark/rapids/shims/RegExpShim.scala:21]: Follow up issue: https://github.com/NVIDIA/spark-rapids/issues/7105

**Comment by firestarman** [comment]
build

**Pull request merged**